protocols/util/factories/avro_factory.py

```diff
@@ -107,7 +109,7 @@ def mock_complex_type(field, field_type, dependencies, version, fill_nullables):
     """
     value = None
     if field_type == 'record':
-        if isinstance(field.type, unicode):
+        if isinstance(field.type, str) or isinstance(field.type, basestring):
```

Comment:
are we sure about `unicode` not needing to be here?

Reply:
Yes, unicode has been renamed string in python 3 but we use it in python 2. There is a basestring type from past.builtins that evaluates to the right thing dependent on version.